### PR TITLE
Add support for payment_intent on invoice and default_payment_method

### PR DESCRIPTION
**Pull request by remi-stripe**
This is now ready to be released. Those fields will be public later this week and are stable. Having this out ahead of time is likely the best approach.

r? @ob-stripe 
cc @stripe/api-libraries

### CODE REVIEW

File: src/Stripe.net/Services/Invoices/InvoiceUpdateOptions.cs [hunk 1]
```diff
@@ -32,6 +32,12 @@ public class InvoiceUpdateOptions : BaseOptions
         [JsonProperty("days_until_due")]
         public long? DaysUntilDue { get; set; }
 
+        /// <summary>
+        /// ID of the default payment method for the invoice.
+        /// </summary>
+        [JsonProperty("custom_fields")]
+        public string DefaultPaymentMethodId { get; set; }
```
Comment (mickjermsurawong-stripe) [inline src/Stripe.net/Services/Invoices/InvoiceUpdateOptions.cs]: @remi-stripe  the `JsonProperty` should be `default_payment_method` here?

File: src/Stripe.net/Services/Subscriptions/SubscriptionSharedOptions.cs [hunk 2]
```diff
@@ -50,6 +50,12 @@ public abstract class SubscriptionSharedOptions : BaseOptions
         [JsonProperty("days_until_due")]
         public long? DaysUntilDue { get; set; }
 
+        /// <summary>
+        /// ID of the default payment method for the subscription.
+        /// </summary>
+        [JsonProperty("custom_fields")]
+        public string DefaultPaymentMethodId { get; set; }
```
Comment (mickjermsurawong-stripe) [inline src/Stripe.net/Services/Subscriptions/SubscriptionSharedOptions.cs]: Same here on JsonProperty `default_payment_method`

File: src/Stripe.net/Services/Invoices/InvoiceUpdateOptions.cs [hunk 3]
```diff
@@ -32,6 +32,12 @@ public class InvoiceUpdateOptions : BaseOptions
...
+
```
Comment (mickjermsurawong-stripe) [inline src/Stripe.net/Services/Invoices/InvoiceUpdateOptions.cs]: Same as in Go. We missed `payment_method` in https://github.com/stripe/stripe-dotnet/blob/master/src/Stripe.net/Services/Invoices/InvoicePayOptions.cs#L8

**Comment by mickjermsurawong-stripe** [comment]
ptal @remi-stripe

**Comment by remi-stripe** [comment]
@mickjermsurawong-stripe Nice catch on the mistakes. I had no idea `payment_method` was allowed on Invoice Pay either. I fixed all of them I think

File: src/Stripe.net/Entities/Customers/CustomerInvoiceSettings.cs [hunk 4]
```diff
@@ -11,6 +12,32 @@ public class CustomerInvoiceSettings : StripeEntity
         [JsonProperty("custom_fields")]
         public List<InvoiceCustomField> CustomFields { get; set; }
 
+        #region Expandable DefaultPaymentMethod
+
+        /// <summary>
+        /// ID of the default payment method for the customer.
+        /// </summary>
+        public string DefaultPaymentMethodId { get; set; }
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/Customers/CustomerInvoiceSettings.cs]: Can you add a `[JsonIgnore]` annotation above please?

File: src/Stripe.net/Entities/Invoices/Invoice.cs [hunk 5]
```diff
@@ -103,6 +103,35 @@ internal object InternalCustomer
...
 
+        #region Expandable DefaultPaymentMethod
+
+        /// <summary>
+        /// ID of the default payment method for the invoice. It must belong to the customer
+        /// associated with the invoice and be in a chargeable state. If not set, defaults to the
+        /// subscription’s default payment method, if any, or to the customer’s default payment
+        /// method.
+        /// </summary>
+        public string DefaultPaymentMethodId { get; set; }
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/Invoices/Invoice.cs]: Same here.

File: src/Stripe.net/Entities/Subscriptions/Subscription.cs [hunk 6]
```diff
@@ -90,6 +90,32 @@ internal object InternalCustomer
         [JsonProperty("days_until_due")]
         public long? DaysUntilDue { get; set; }
 
+        #region Expandable DefaultPaymentMethod
+
+        /// <summary>
+        /// ID of the default payment method for the subscription.
+        /// </summary>
+        public string DefaultPaymentMethodId { get; set; }
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/Subscriptions/Subscription.cs]: And here :)

**Comment by remi-stripe** [comment]
@ob-stripe Fixed, PTAL!

**Pull request merged**